README.md

```diff
@@ -376,12 +376,139 @@ export class VehicleProductComponent extends NgxSubFormRemapComponent<OneVehicle
+    value: ArrayTypeOfPropertyOf<CrewMembersForm>,
+  ): FormControl {
+    switch (key) {
+      // note: the following string is type safe based on your form properties!
+      case 'crewMembers':
+        return new FormControl(value, [Validators.required]);
+      default:
+        return new FormControl(value);
+    }
+  }
+}
+```
+
+Then our view will look like the following:
+
+```html
+<!-- src/app/main/listing/listing-form/vehicle-listing/crew-members/crew-members.component.html#L1-L26 -->
+
+<fieldset [formGroup]="formGroup" class="container">
+  <legend>Crew members form</legend>
+
+  <div
+    class="crew-member"
+    formArrayName="crewMembers"
+    *ngFor="let crewMember of formGroupControls.crewMembers.controls; let i = index"
+  >
+    <app-crew-member [formControl]="crewMember"></app-crew-member>
+
+    <button mat-mini-fab color="primary" (click)="removeCrewMember(i)" [disabled]="formGroup.disabled">
+      <mat-icon>delete</mat-icon>
+    </button>
+  </div>
+
+  <button
+    mat-raised-button
+    data-btn-add-crew-member
+    color="primary"
+    class="add-crew-member"
+    (click)="addCrewMember()"
+    [disabled]="formGroup.disabled"
+  >
+    Add a crew member
+  </button>
+</fieldset>
+```
+
+The `app-crew-member` component is a simple `NgxSubFormComponent` as you can imagine:
+
+```ts
+// src/app/main/listing/listing-form/vehicle-listing/crew-members/crew-member/crew-member.component.ts#L6-L19
+
+@Component({
+  selector: 'app-crew-member',
+  templateUrl: './crew-member.component.html',
+  styleUrls: ['./crew-member.component.scss'],
+  providers: subformComponentProviders(CrewMemberComponent),
+})
+export class CrewMemberComponent extends NgxSubFormComponent<CrewMember> {
+  protected getFormControls(): Controls<CrewMember> {
+    return {
+      firstName: new FormControl(null, [Validators.required]),
+      lastName: new FormControl(null, [Validators.required]),
+    };
+  }
+}
+```
 
 ### Helpers
 
 - `onFormUpdate` hook: Allows you to react whenever the form is being modified. Instead of subscribing to `this.formGroup.valueChanges` or `this.formControls.someProp.valueChanges` you will not have to deal with anything asynchronous nor have to worry about subscriptions and memory leaks. Just implement the method `onFormUpdate(formUpdate: FormUpdate<FormInterface>): void` and if you need to know which property changed do a check like the following: `if (formUpdate.yourProperty) {}`. Be aware that this method will be called only when there are either local changes to the form or changes coming from subforms. If the parent `setValue` or `patchValue` this method won't be triggered
 - `getFormGroupControlOptions` hook: Allows you to define control options for construction of the internal FormGroup. Use this to define form-level validators
+- `createFormArrayControl` hook: Allows you to create the `FormControl` of a given property of your form (to define validators for example)
```

Comment:
Missing: Add a small text to say that it needs to implements the interface first